### ISSUE TITLE
HTTPS Server gets killed when scanning with CryptCheck

### ISSUE THREAD

**Issue by denniskaselow** [dart-lang/sdk#29307]
I've got an HTTPS-Server using HttpServer.bindSecure with a Let's Encrypt certificate.

Running
Dart VM version: 1.23.0-dev.11.6 (Fri Apr  7 08:27:41 2017) on "linux_x64"
on
Linux 3.16.0-4-amd64 x86_64
Debian GNU/Linux 8.7 (jessie)

If I use CryptCheck (https://tls.imirhil.fr/) targeted at my server, it immediately gets killed and dumps the following to the console:
```
Dumping native stack trace for thread 4cb6
  [0x00007f607f36b935] strrchr
  [0x00007f607f36b935] strrchr
  [0x0000000000547209] Unknown symbol
  [0x00000000005468b8] dart::bin::SSLFilter::Handshake()
  [0x0000000000546832] dart::bin::Builtin_SecureSocket_Handshake(_Dart_NativeArguments*)
  [0x00000000007e3b17] dart::NativeEntry::NativeCallWrapperNoStackCheck(_Dart_NativeArguments*, void (*)(_Dart_NativeArguments*))
  [0x00007f608056e7a3] [Stub] CallNativeCFunction
  [0x00007f607ef8a7c7] _SecureFilterImpl.handshake
  [0x00007f607ef8a604] _RawSecureSocket._secureHandshake
  [0x00007f607d654673] _RawSecureSocket._closeHandler
  [0x00007f607ef95719] _RawSecureSocket._eventDispatcher
  [0x00007f607ef95500] _RawSecureSocket._eventDispatcher
  [0x00007f607d658c3d] _Closure.call
  [0x00007f607ec75ec9] _rootRunUnary
  [0x00007f607ec75d63] _rootRunUnary
  [0x00007f607ec75c6c] _Closure.call
  [0x00007f607ec75acf] _CustomZone.runUnary
  [0x00007f607ec80cd2] _CustomZone.runUnaryGuarded
  [0x00007f607ef950af] _BufferingStreamSubscription._sendData
  [0x00007f607ef94d90] _BufferingStreamSubscription._add
  [0x00007f607ef94c1f] _StreamController&&_SyncStreamControllerDispatch._sendData
  [0x00007f607ef94a31] _StreamController._add
  [0x00007f607ef94744] _StreamController.add
  [0x00007f607d65420d] _RawSocket._RawSocket.<anonymous closure>
  [0x00007f607e953c5d] _Closure.call
  [0x00007f607ef8edb9] _NativeSocket.issueReadEvent.issue
  [0x00007f607e953c5d] _Closure.call
  [0x00007f607c589ac6] _rootRun
  [0x00007f607c589980] _rootRun
  [0x00007f607c589889] _Closure.call
  [0x00007f607c58934c] _CustomZone.run
  [0x00007f607c58914f] _CustomZone.runGuarded
  [0x00007f607c58eed5] _CustomZone.bindCallback.<anonymous closure>
  [0x00007f607e953c5d] _Closure.call
  [0x00007f607c589b2e] _rootRun
  [0x00007f607c589980] _rootRun
  [0x00007f607c589889] _Closure.call
  [0x00007f607c58934c] _CustomZone.run
  [0x00007f607c58914f] _CustomZone.runGuarded
  [0x00007f607c58eed5] _CustomZone.bindCallback.<anonymous closure>
  [0x00007f607e953c5d] _Closure.call
  [0x00007f607e95d31d] _microtaskLoop
  [0x00007f607e95cec0] _startMicrotaskLoop
  [0x00007f607e95cd94] _startMicrotaskLoop
  [0x00007f607e953c5d] _Closure.call
  [0x00007f607e95cc78] _runPendingImmediateCallback
  [0x00007f607e958ebf] _RawReceivePortImpl._handleMessage
  [0x00007f608056e9d9] [Stub] InvokeDartCode
  [0x00000000006ddc40] dart::DartEntry::InvokeFunction(dart::Function const&, dart::Array const&, dart::Array const&, unsigned long)
  [0x00000000006e0e85] dart::DartLibraryCalls::HandleMessage(dart::Object const&, dart::Instance const&)
  [0x00000000007ae957] dart::IsolateMessageHandler::HandleMessage(dart::Message*)
  [0x00000000007d6716] dart::MessageHandler::HandleMessages(dart::MonitorLocker*, bool, bool)
  [0x00000000007d6be4] dart::MessageHandler::TaskCallback()
-- End of DumpStackTrace
```

This does not happen if I use the stable version:
Dart VM version: 1.22.1 (Tue Feb 21 22:22:42 2017) on "linux_x64"

**Comment by zanderso** [comment]
The logic here https://github.com/dart-lang/sdk/blob/master/runtime/bin/secure_socket_boringssl.cc#L92 is new and looks a little suspicious, and that's probably the only place we're calling `strrchr()`. @aarongreen

**Comment by zanderso** [comment]
This crasher was introduced after 1.22, so I hope there's still time to get the fix in for 1.23.
